kuksa_go_client/kuksa_client/grpc.go

```diff
@@ -304,8 +311,30 @@ func (cg *KuksaClientCommGrpc) UnsubscribeFromKuksaVal(id string) error {
 }
 
 // Function to authorize to kuksa.val server
-func (cg *KuksaClientCommGrpc) AuthorizeKuksaValConn() error {
-	log.Println("KUKSA.val databroker does not support authorization mechanism yet. If you want to use authorization please use KUKSA.val server")
+func (cg *KuksaClientCommGrpc) AuthorizeKuksaValConn(TokenOrTokenfile string) error {
+	var tokenString string
+	if TokenOrTokenfile == "" {
+		if cg.Config.TokenOrTokenfile != ""{
```

Comment:
Would it be an idea to reverse the logic here

if cg.Config.TokenOrTokenfile != ""
  <use the file from config file>
else
  <use the hardcoded string from TokenOrTokenFile>

Then we have no "dead" code, and one could actually use the config in https://github.com/eclipse/kuksa.val/blob/master/kuksa_go_client/kuksa-client.json (which by the way needs a changed keyword for "token", and also another port if the default is for a kuksa-val server connection (port 8090)